### PR TITLE
Patch signature

### PR DESCRIPTION
**Pull request by benjaminbollen**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/533)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @inetic

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by benjaminbollen** [comment]
patch ready for review

File: src/routing_membrane.rs [hunk 1]
```diff
@@ -502,7 +496,8 @@ impl<F> RoutingMembrane<F> where F: Interface {
         }
 
         // Forward
-        ignore(self.send_swarm_or_parallel_or_relay(&message));
+        ignore(self.send_swarm_or_parallel_or_relay_with_signature(
+            &message, message_wrap.signature().clone()));
```
Comment (inetic) [inline src/routing_membrane.rs]: Could this new `send_swarm_or_parallel_or_relay_with_signature` function just take `SignedMessage` as argument? In this case the `message_wrap` variable.

Comment (benjaminbollen) [inline src/routing_membrane.rs]: I went for this approach - fully in agreement it is not the most elegant - as it is clear, correct and consistent with the approach where no destination is passed into the send_* function (from handlers side) as to make the destination always read from the RoutingMessage;

equally taking into account that no additional unwrapping of the RoutingMessage needs to occur.

**Pull request merged**